java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture10/tasks/_1/NoFileException.java

```diff
@@ -0,0 +1,7 @@
+package com.zoolatech.lecture10.tasks._1;
+
+final class NoFileException extends Error {
```

Comment:
2 thoughts on that:
Error is a very specific type of erorrs. Let's say that we are not talking about checked exceptions.  You have option to define custom exception which is unchecked (RuntimeException) or to use Error. The main difference is that if a client of this code might want to catch it - use unchecked one. If client cannot fix this problem - use Error (extremly rarely used). Common example of errors are OutOfMemoryError (it means that client cannot fix the problem. you don't have memory and you cannot allocate it in runtime). So in your case extends RuntimeException would be better. But I'm strongly belive that it's better to avoid custom exception if you have one in jdk and don't need special handling. I'd use FileNotFoundException which is part of jdk.